### PR TITLE
PMM-11842 BM enabled by default upgrade tests 

### PR DESCRIPTION
**Pull request by vyarosh**

### CODE REVIEW

File: .github/workflows/pmm-ui-tests-matrix-upgrade.yml [hunk 1]
```diff
@@ -51,3 +51,19 @@ jobs:
...
         testing_repository: ${{ inputs.testing_repository }}
         experimental_repository: ${{ inputs.experimental_repository }}
+
+   general:
+     name: General
+     uses: ./.github/workflows/pmm-ui-tests-upgrade.yml
+     secrets: inherit
+     with:
+       pmm_ui_tests_branch: ${{ inputs.pmm_ui_tests_branch || 'main' }}
+       pmm_pre_upgrade_flag: '@pre-upgrade'
```
Comment (peterSirotnak) [inline .github/workflows/pmm-ui-tests-matrix-upgrade.yml:61]: We talked about using features flags for example @backup-pre-upgrade @backup-post-upgrade

File: playwright-tests/api/serverApiV1.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import apiHelper from "@api/helpers/apiHelper";
+import Duration from "@helpers/Duration";
+
+export const serverAPIv1 = {
```
Comment (peterSirotnak) [inline playwright-tests/api/serverApiV1.ts:4]:
```suggestion
export const serverAPI = {
```

File: playwright-tests/api/settingsApiV1.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import {APIResponse} from "playwright-core";
+
+const PATH_GET = 'v1/Settings/Get';
+const PATH_CHANGE = 'v1/Settings/Get';
+
+export enum SettingProperty {
+  bm = "backup_management_enabled",
+}
+
+export const settingsAPIv1 = {
```
Comment (peterSirotnak) [inline playwright-tests/api/settingsApiV1.ts:11]:
```suggestion
export const settingsAPI = {
```

File: playwright-tests/tests/upgrade/upgradePmm.spec.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    await homeDashboard.pmmUpgrade.verifyUpgradeWidget();
+  });
+
+  test('PMM-T3 Verify user is able to Upgrade PMM version [blocker] @pmm-upgrade @ovf-upgrade @ami-upgrade', async ({ page }) => {
+    const homeDashboard = new HomeDashboard(page);
+
+    await homeDashboard.upgradePMM();
+  });
+
+  if (pmmVersion < 36) {
```
Comment (peterSirotnak) [inline playwright-tests/tests/upgrade/upgradePmm.spec.ts:37]: Move this as we talked about to its folder and spec file PMM-T1659 both tests

**Review by peterSirotnak — CHANGES_REQUESTED** [review]

File: .github/workflows/pmm-ui-tests-matrix-upgrade.yml (hunk 1 above)
Comment (vyarosh) [reply .github/workflows/pmm-ui-tests-matrix-upgrade.yml:61]: Yes and these tests are exceptions which have to be run separately from it's feature "BM", but default settings check does not deserve a dedicated feature at least for now. Yet we will have test which will be classified as general checks or will cross multiple features and will require a dedicated group. So I suppose here is our start with "General" group and "RBAC" already have own group.

File: playwright-tests/tests/upgrade/upgradePmm.spec.ts (hunk 4 above)
Comment (vyarosh) [reply playwright-tests/tests/upgrade/upgradePmm.spec.ts]: Which one feature do you think it is? Asking as I don't think this is a dedicated feature.

File: playwright-tests/api/settingsApiV1.ts (hunk 3 above)
Comment (vyarosh) [reply playwright-tests/api/settingsApiV1.ts:11]: Why do you want to exclude the API version from the file name? This important detail shows given API origin - this is a client facing api and its' version is 1

File: playwright-tests/api/serverApiV1.ts (hunk 2 above)
Comment (vyarosh) [reply playwright-tests/api/serverApiV1.ts:4]: Please see the following comment

File: .github/workflows/pmm-ui-tests-matrix-upgrade.yml (hunk 1 above)
Comment (vyarosh) [reply .github/workflows/pmm-ui-tests-matrix-upgrade.yml]: moved to "configuration" feature

File: playwright-tests/api/serverApiV1.ts (hunk 2 above)
Comment (vyarosh) [reply playwright-tests/api/serverApiV1.ts]: fixed

File: playwright-tests/api/settingsApiV1.ts (hunk 3 above)
Comment (vyarosh) [reply playwright-tests/api/settingsApiV1.ts]: fixed

File: playwright-tests/tests/upgrade/upgradePmm.spec.ts (hunk 4 above)
Comment (vyarosh) [reply playwright-tests/tests/upgrade/upgradePmm.spec.ts]: moved to "configuration" feature

File: .github/workflows/pmm-ui-tests-upgrade.yml [hunk 5]
```diff
@@ -135,24 +135,23 @@ jobs:
...
 
       - name: Install NodeJS v16
-        uses: percona-platform/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: 16.14.1
 
       - name: Checkout UI tests
-        uses: percona-platform/checkout@v2
+        uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (vyarosh) [reply .github/workflows/pmm-ui-tests-upgrade.yml]: moved out to the separate PR

**Comment by vyarosh** [comment]
https://github.com/percona/pmm-ui-tests/actions/runs/4559645314/jobs/8043813357
https://github.com/percona/pmm-ui-tests/actions/runs/4559955307/jobs/8044429470

**Pull request merged**